test/async_test.js

```diff
@@ -0,0 +1,165 @@
+/*globals fetch, async */
+"use strict";
+describe("async API tests", () => {
+  it("should asynchronously resolve to the string pass.", () => {
+    var test = async(function* () {
+      var x = yield Promise.resolve("pass");
+      return x;
+    });
+    return test().should.become("pass");
+  });
+
+  it("should reject when rejected.", () => {
+    var error = new Error("testing111");
+    var test = async(function* () {
+      throw error;
+    });
+    return test().should.eventually.be.rejectedWith(error);
+  });
+
+  it("should recover and return in a catch.", () => {
+    var error = new Error("");
+    var test = async(function* () {
+      try {
+        yield Promise.reject(error);
+      } catch (err) {
```

Comment:
Had forgotten this in the original code :blush: ... who would have thought unit testing would find bugs!